### ISSUE TITLE
Modify the existing nginx configuration, including an additional nginx module

### ISSUE THREAD

**Issue by AYEG** [tiangolo/full-stack#20]
Hi!
I'm looking to try out a different configuration for nginx to see if it can resolve a rpc connection problem and came across this section on custom nginx configurations
https://github.com/tiangolo/uwsgi-nginx-flask-docker#customizing-nginx-configurations

I've created a new file nginx_rpc_location_pass.conf that is found when starting the backend container. 
It just has this:
server { 
    listen 80 http2;
    location / {
         grpc_pass grpc: alpha_1: 9080;
    }
 }
According to the nginx docs I would need to use ngx_http_v2_module for this directive. 

How can include that module in the nginx? Is there a way to include that at a later stage or should I actually rebuild the base image and import that into the backend.dockerfile. 

I'm using the appropriate traefik grpc labels and traefik v1.7 with the latest grpc dependencies.  I'm not  sure if what I'm trying makes sense, background:

 I'm trying to communicate to a Dgraph database from the backend container and after a period of inactivity the query from the client returns:
"<_Rendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "OS Error"
	debug_error_string = "{"created":"@1546772849.163482552","description":"Error received from peer","file":"src/core/lib/surface/call.cc","file_line":1036,"grpc_message":"OS Error","grpc_status":14}"
>"
After retrying the same query immediately it does work like it should, if possible, I'd like to prevent this connection error from happening all together. (Also tried changing host tcp settings to recommended values with no effect)

**Comment by AYEG** [comment]
It turns out I was reading outdated information, the http_v2_module has long been included in the stable release of Nginx. To see which modules can be configured in the current Nginx version:

> docker exec -it <backend_container_id> bash
> nginx -V 

Loading the module in the extra nginx .conf doesn't work
` "load_module" directive is not allowed"`

"To enable the module it needs to be included as arguments during the installation as it is not built by default ", http://nginx.org/en/docs/http/ngx_http_v2_module.html  
using the --with-http_v2_module configuration parameter

So it seems I need to add a line somewhere here
https://github.com/tiangolo/uwsgi-nginx-docker/blob/master/python3.6/Dockerfile

**Comment by AYEG** [comment]
integrating the uwsgi-nginx-flask base images (and nginx.conf) into the project can be done as follows:
- include new dockerfiles for uwsgi-nginx and uwsgi-nginx-flask
- include build commands 
- include depends_on under backend in d-c.deploy.images.yml and d-c.shared.depends.yml:
- uwsgi-nginx 
- uwsgi-nginx-flask
https://github.com/tiangolo/uwsgi-nginx-docker (pick the correct python version)
copy uwsgi.ini nginx.conf and supervisord.conf entrypoint.sh into the project dir into backend folder
https://github.com/tiangolo/uwsgi-nginx-flask-docker  (pick the correct python version)
copy the sh scripts 

Finally change the FROM lines in uwsgi-nginx-flask.dockerfile and backend.dockerfile

Now it should be easy to include some lines into the uwsgi-nginx dockerfiles like "&& CONFIG="  to compile nginx differently.

**Comment by tiangolo** [comment]
Sorry for the delay @AYEG, I see you solved your problem. Nevertheless, I recently added a feature to allow overriding completely all the Nginx configuration, the new docs are here: https://github.com/tiangolo/uwsgi-nginx-flask-docker#overriding-nginx-configuration-completely

Also, on another note, I created https://github.com/tiangolo/fastapi, inspired by Flask, many of the plug-ins from this project generator and other tools.

If you are using this project generator with all the plug-ins, you would probably enjoy FastAPI a lot.

There is a also a comparable project generator to this one (but with more featuers): https://github.com/tiangolo/full-stack-fastapi-postgresql
